extensions/aws/src/main/java/io/opentelemetry/extension/aws/AwsXrayPropagator.java

```diff
@@ -180,8 +196,12 @@ public static AwsXrayPropagator getInstance() {
         spanId = parseSpanId(value);
       } else if (trimmedPart.startsWith(SAMPLED_FLAG_KEY)) {
         isSampled = parseTraceFlag(value);
+      } else {
+        if (baggage == null) {
+          baggage = Baggage.builder();
+        }
+        baggage.put(trimmedPart.substring(0, equalsIndex), value.trim());
```

Comment:
So interestingly the X-Ray Trace header spec has a limit of 256 bytes on incoming arbitrary key-value pairs after the defined fields (baggage). This is not publicly documented, and is only implemented in the [Node SDK](https://github.com/aws/aws-xray-sdk-node/blob/b88909ef98c1b29c82691ced4153d1693c872c8f/packages/core/lib/utils.js#L194). So clearly it's not critical to matching the behavior of the X-Ray SDKs, but I think it's still worth calling out since a limit on the baggage size is a safe thing to do to prevent malicious headers from stalling a program. Would be interested to hear your thoughts/if there are any existing considerations for this in OTel.

Reply:
Thanks for calling this out! The interesting point is to compare to w3c baggage

https://w3c.github.io/baggage/#limits

It's limit is 8192 - it's different but I don't see a big reason for that to affect how xray propagator itself works given it's a different spec. And smaller can be increased to larger later if we ever needed to. Will add the limit.

Reply:
Awesome! Good to know about the w3c limits. LGTM.